internal/util/stringheap.go

```diff
@@ -0,0 +1,28 @@
+package util
+
+// StringHeap is an array of strings that implements container/heap.Interface.
+type StringHeap []string
```

Comment:
Would it be difficult to make the heap contain the `ingressTranslationMeta` (just as in the `map`) to prevent having both `map` and `heap` inside the `ingressTranslationIndex`?

Reply:
I tried as much and got sad discovering that this didn't really slot neatly into the simple data structure :) We have to modify the entries as we go to add new paths, and we find the correct entry to modify based on the key, so we basically end up with some combination of a map and a heap to manage that, sooooo...